src/Common/src/System/Security/Cryptography/DSACng.ImportExport.cs

```diff
@@ -0,0 +1,377 @@
+                if (keySizeInBits <= MaxV1KeySize)
+                {
+                    GenerateV1DsaBlob(out blob, parameters, keySizeInBytes, hasPrivateKey);
+                }
+                else
+                {
+                    GenerateV2DsaBlob(out blob, parameters, keySizeInBytes, hasPrivateKey);
+                }
+
+                ImportKeyBlob(blob, hasPrivateKey);
+            }
+
+            private static void GenerateV1DsaBlob(out byte[] blob, DSAParameters parameters, int cbKey, bool includePrivate)
+            {
+                // We need to build a key blob structured as follows:
+                //
+                //     BCRYPT_DSA_KEY_BLOB       header
+                //     byte[cbKey]               P
+                //     byte[cbKey]               G
+                //     byte[cbKey]               Y
+                //     -- Private only --
+                //     byte[Sha1HashOutputSize]  X
+
+                unsafe
+                {
+                    int blobSize = 
+                        sizeof(BCRYPT_DSA_KEY_BLOB) +
+                        cbKey +
+                        cbKey +
+                        cbKey;
+
+                    if (includePrivate)
+                    {
+                        blobSize += Sha1HashOutputSize;
+                    }
+
+                    blob = new byte[blobSize];
+                    fixed (byte* pDsaBlob = blob)
+                    {
+                        // Build the header
+                        BCRYPT_DSA_KEY_BLOB* pBcryptBlob = (BCRYPT_DSA_KEY_BLOB*)pDsaBlob;
+
+                        pBcryptBlob->Magic = includePrivate ? KeyBlobMagicNumber.BCRYPT_DSA_PRIVATE_MAGIC : KeyBlobMagicNumber.BCRYPT_DSA_PUBLIC_MAGIC;
+                        pBcryptBlob->cbKey = cbKey;
+
+                        int offset = sizeof(KeyBlobMagicNumber) + sizeof(int); // skip Magic and cbKey
+
+                        if (parameters.Seed != null)
+                        {
+                            // The Seed length is hardcoded into BCRYPT_DSA_KEY_BLOB, so check it now we can give a nicer error message.
+                            if (parameters.Seed.Length != Sha1HashOutputSize)
+                                throw new ArgumentException(SR.Cryptography_InvalidDsaParameters_SeedRestriction_ShortKey);
+
+                            Interop.BCrypt.Emit(blob, ref offset, ToBigEndian(parameters.Counter));
+                            Interop.BCrypt.Emit(blob, ref offset, parameters.Seed);
+                        }
+                        else
+                        {
+                            // If Seed is not present, back fill both counter and seed with 0xff. Do not use parameters.Counter as CNG is more strict than CAPI and will reject
+                            // anything other than 0xffffffff. That could complicate efforts to switch usage of DSACryptoServiceProvider to DSACng.
+                            byte[] ff = { 0xff };
```

Comment:
How common is this code path? Could it be done in a way that avoids the temp `byte[]` allocation (i.e. maybe a `BCrypt.Emit` overload that takes a single `byte` value to emit, or single `byte` value plus `int count` number of times to emit that single byte)?